### ISSUE TITLE
When using the GUI to edit automations, "drag and drop" will sometimes make steps just "disappear"

### ISSUE THREAD

**Issue by VernGraner** [home-assistant/frontend#23171]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

The problem is intermittent so difficult to recreate. That said, I was able to get video of it happening. Essentially, you duplicate an item, then try to drag it to a new location. When you "drop" it, rather than landing where you placed it, the item simply disappears.

https://github.com/user-attachments/assets/2fe610f2-845d-48fa-a31a-0317aa3554ae

### Describe the behavior you expected

When you drop a line, it should appear ¯\_(ツ)_/¯ 

### Steps to reproduce the issue

1.Duplicate and item in the automation
2. drag it to a new location
3. drop it
...

### What version of Home Assistant Core has the issue?

2024.12.0

### In which browser are you experiencing the issue with?

Chrome Version 131.0.6778.108 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Windows 11 24H2 OS Build 26100.2454

### State of relevant entities

```txt
The issue is intermittent but I have had it happen on Windows 11 as well as on an Android Tablet.
```

**Comment by MichaelMKKelly** [comment]
I have also seen this issue, I think its something to do with if something is dropped outside of somewhere it belongs. and its pretty specific so if your quick or slightly off you can drop it where its not supposed to go and it disappears.

perhaps on invalid placement it should be put back where it was and show a warning "oops - that doesn't go there" or something.
Or an undo button might be cool but that's probably a lot more work.

**Comment by arifroni** [comment]
i have experienced this also and i am sure i didn't missed the right spot to drop. i have to save automation after every successful drag and drop. its pretty annoying.

**Comment by jpbaril** [comment]
> I think its something to do with if something is dropped outside of somewhere it belongs. and its pretty specific so if your quick or slightly off you can drop it where its not supposed to go and it disappears.

No. I'm just moving a block out of another block and it still disappears even if the moved out action can for sure belong where I dropped it.

https://github.com/user-attachments/assets/7f999248-ba6d-4825-9dc1-4f0bacd095e7

**Comment by zekone-boar** [comment]
I am also having this issue; running HAOS 2025.5.2, accessing via Chrome, issue present since at least start of 2025.

As noted by others, I have seem to have narrowed it down to when moving items into, out of, or between some kind of group - a if/then, a choose, a run in sequence, that kind of thing.

Also, I noticed that when I drag an item off the action list, the cursor changes to: 🚫. If I let go of an item with the 🚫 cursor visible, the item does not vanish, it either remains where it was or moves to the closest visible point it can reach. _This is all expected behaviour_. However with this bug, the cursor remains the same regular system pointer, so you aren't even aware before you let go of the mouse button that there will be a problem. 

Below video shows the issue.

- At 9s, I successfully move the "Automation turn off" action into the "If/then" group.
- Then at 12s I try to move the "Automation turn off" action from the "If/then" group into the "Run in sequence" group at the bottom of the automation, and that is where this bug happens.
- The "Automation turn off" action vanishes: it doesn't move into the "Run in sequence" group, remain in the "If/then" group, or end up elsewhere in the automation.
- To show the expected behaviour I described in the last paragraph, at 20s I demonstrate how when moving an action off the automation list the cursor changes to 🚫. When I let go of the mouse button, the action being dragged remains in the list (it ends up at the bottom) and doesn't vanish.

https://github.com/user-attachments/assets/25d7e0bc-ee6e-4e8b-89c1-1c3d36708de3

**Comment by choenig** [comment]
Just to add:

I can reproduce this bug in Arc-Browser (Chrome) and Safari, but *not* in Firefox.

**Comment by caitotheonlypotato** [comment]
> I have also seen this issue, I think its something to do with if something is dropped outside of somewhere it belongs. and its pretty specific so if your quick or slightly off you can drop it where its not supposed to go and it disappears.
> 
> perhaps on invalid placement it should be put back where it was and show a warning "oops - that doesn't go there" or something. Or an undo button might be cool but that's probably a lot more work.

Not necessarily, for me I've found that you can drag for example some action from a base condition to another condition that's nested in an if / and / not , placing in the relevant action area of the target condition, and it will disappear.  Not every time though, often the first one will work and then subsequent ones fail.
In the interim, I generally now don't take the chance now and revert to cut/paste

**Comment by adynis** [comment]
I am experiencing the same and (at least now speaking) I can reproduce it each time ! (i created a copy of the automation and after ~1h If I do the same like in the video , I have again the problem. Is there anything I could try/test to help debugging the issue ? 
_PS. Sometimes I had the feeling that it is happening when the target place is the last one in the action list_

https://github.com/user-attachments/assets/bcb7ea75-895e-432a-8d69-cd7222334a22

**Comment by VernGraner** [comment]
> I was able to reproduce this a couple times in a dev environment with this automation:
> 
Wow! Thank you for doing this in-depth analysis! 🤯 This seems like real progress or maybe even discovery of "root cause" on a tricky intermittent issue.  Great work! 👍🙂
